whisperv6/whisper.go

```diff
@@ -876,14 +876,18 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 				log.Warn("failed to decode envelopes, peer will be disconnected", "peer", p.peer.ID(), "err", err)
 				return errors.New("invalid envelopes")
 			}
-
 			trouble := false
 			for _, env := range envelopes {
 				cached, err := whisper.add(env, whisper.LightClientMode())
 				if err != nil {
 					trouble = true
 					log.Error("bad envelope received, peer will be disconnected", "peer", p.peer.ID(), "err", err)
 				}
+				whisper.envelopeFeed.Send(EnvelopeEvent{
```

Comment:
What about https://github.com/status-im/whisper/blob/ce8adebb4984a8a9a9d438aaf5159b10fc2cfe7c/whisperv6/whisper.go#L1198-L1201? This event is not useful due to some reason? If that's the case, I would add a comment which would explain that.

Reply:
i need to know peer that delivered an envelope and event should be sent to the feed even if this envelope already in the local cache. that is why i won't be able to already available event.

indeed this is now a bit confusing, will add a reason for new event to the code.

i will check how we use EventEnvelopeAvailable, maybe it can be removed

Reply:
so the only place where EventEnvelopeAvailable is used is in tests and in node canary. I guess we can use same event with slight refactoring.

do you know about other places where this event is used?

Reply:
No, I just remember that a similar thing exists but if we can get rid of it and only keep this new one, which is more versatile, let's do that!

Reply:
i realized that it is not possible, that event is added right after filters are updated with a message on purpose. i cant achieve same semantics so i will keep both for now